src/controllers/userControllers/updateUser.js

```diff
@@ -0,0 +1,30 @@
+const Users = require('../../database');
+
+function updateUser(req, res) {
+  const { email, name, oldPassword, newPassword } = req.body;
+  // console.log(req.params.id);
+  let id = req.params.id;
+  id = Users.findIndex((user) => {
+    return user.id == id;
+  });
+  if (id === -1) {
+    res.status(404).send({ message: 'Not Found' });
+    return;
+  }
+
+  // console.log(Users);
+  if (oldPassword !== Users[id].password) {
+    res.status(400).send({ message: 'you entered a wrong password' });
+    return;
+  }
```

Comment:
usually when the password doesn't match we return `401` Unauthorized.

Reply:
yea, I missed that one.
fixed!